providers/cstor/cstor-disk-pool/test.yml

```diff
@@ -41,38 +41,91 @@
             line: '     - {{ item }}'
           with_items: "{{ diskList.stdout_lines }}"
 
-        - name: Create cStor Disk Pool
-          shell: kubectl create -f "{{ playbook_dir }}/{{ cstor_spc }}"
-          args:
-            executable: /bin/bash
+        - block: 
 
-        - name: Verify if cStor Disk Pool are created on each node
-          shell: kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }},app=cstor-pool --no-headers -o custom-columns=:status.phase
-          args:
-            executable: /bin/bash
-          register: pool_count
-          until: pool_count.stdout_lines|length <= node_count.stdout_lines|length 
-          retries: 10
-          delay: 60
+            - name: Create cStor Disk Pool
+              shell: kubectl create -f "{{ playbook_dir }}/{{ cstor_spc }}"
+              args:
+                executable: /bin/bash
 
-        - name: Get cStor Disk Pool names to verify the container statuses
-          shell: kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }} -l app=cstor-pool --no-headers -o=custom-columns=NAME:".metadata.name"
-          args:
-            executable: /bin/bash
-          register: cstor_pool_pod
+            - name: Verify if cStor Disk Pool are created on each node
+              shell: > 
+                 kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }}
+                 --no-headers -o custom-columns=:status.phase
+              args:
+                executable: /bin/bash
+              register: pool_count
+              until: pool_count.stdout_lines|length <= node_count.stdout_lines|length 
```

Comment:
This condition will pass even if only one pool is created as it is lesser than the number of nodes. This check has to be done with condition equal to MaxPool, not the number of nodes.